frontend/src/scenes/insights/InsightPageHeader.tsx

```diff
@@ -189,6 +193,27 @@ export function InsightPageHeader({ insightLogicProps }: { insightLogicProps: In
                                                 </>
                                             )}
 
+                                            {isInsightVizNode(query) ? (
+                                                <LemonButton
+                                                    status="stealth"
+                                                    onClick={() => {
+                                                        // for an existing insight in view mode
+                                                        if (hasDashboardItemId && insightMode !== ItemMode.Edit) {
+                                                            // enter edit mode
+                                                            setInsightMode(ItemMode.Edit, null)
+
+                                                            // exit early if query editor doesn't need to be toggled
+                                                            if (showQueryEditor !== false) {
+                                                                return
+                                                            }
+                                                        }
+                                                        toggleQueryEditorPanel()
+                                                    }}
+                                                    fullWidth
+                                                >
+                                                    {showQueryEditor ? 'Hide source' : 'View source'}
```

Comment:
We typically keep the last section just for "Delete" since that's such a major action. So I propose putting this directly below the export button in that section, rather than in Delete's section